src/types/FateString.js

```diff
@@ -10,22 +11,44 @@ const toString = (data) => {
     return data.toString()
 }
 
+const toBytes = (data) => {
+    if (typeof data === 'string') {
+        const encoder = new TextEncoder()
+        return encoder.encode(data)
+    }
+
+    return data
+}
+
+const isUnicodeString = (data) => {
+    const bytes = toBytes(toString(data))
+    return byteArray2Int(data) === byteArray2Int(bytes)
+}
+
 class FateString extends FateData {
     constructor(value) {
         super('string')
 
-        this._value = toString(value)
+        this._value = toBytes(value)
     }
 
     get type() {
         return FateTypeString()
     }
 
     toString() {
+        return toString(this._value)
+    }
+
+    toBytes() {
         return this._value
     }
 
     valueOf() {
+        if (isUnicodeString(this._value)) {
+            return this.toString()
+        }
+
         return this._value
```

Comment:
This is the most tricky part, I am not sure how to make it without breaking changes